File: src/main/java/com/vjiki/music/entity/ChatParticipant.java
package com.vjiki.music.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.OffsetDateTime;
import java.util.UUID;

@Entity
@Table(name = "chat_participants", schema = "music")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@IdClass(ChatParticipantId.class)
public class ChatParticipant {

    @Id
    @Column(name = "chat_id", nullable = false)
    private UUID chatId;

    @Id
    @Column(name = "user_id", nullable = false)
    private UUID userId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_id", insertable = false, updatable = false)
    private Chat chat;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(name = "role")
    private ParticipantRole role = ParticipantRole.MEMBER;

    @CreationTimestamp
    @Column(name = "joined_at", nullable = false, updatable = false)
    private OffsetDateTime joinedAt;

    @Column(name = "is_muted", nullable = false)
    private Boolean isMuted = false;

    @Column(name = "last_read_message_id")
    private UUID lastReadMessageId;
}

File: src/main/kotlin/com/vjiki/music/entity/ChatParticipant.kt
package com.vjiki.music.entity

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import java.time.OffsetDateTime
import java.util.*

@Entity
@Table(name = "chat_participants", schema = "music")
@IdClass(ChatParticipantId::class)
data class ChatParticipant(
    @Id
    @Column(name = "chat_id", nullable = false)
    val chatId: UUID,

    @Id
    @Column(name = "user_id", nullable = false)
    val userId: UUID,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "chat_id", insertable = false, updatable = false)
    val chat: Chat? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", insertable = false, updatable = false)
    val user: User? = null,

    @Enumerated(EnumType.STRING)
    @Column(name = "role")
    val role: ParticipantRole = ParticipantRole.MEMBER,

    @CreationTimestamp
    @Column(name = "joined_at", nullable = false, updatable = false)
    val joinedAt: OffsetDateTime? = null,

    @Column(name = "is_muted", nullable = false)
    val isMuted: Boolean = false,

    @Column(name = "last_read_message_id")
    val lastReadMessageId: UUID? = null
)

